src/net_processing.cpp

```diff
@@ -3122,10 +3122,11 @@ bool SendMessages(CNode* pto, CConnman& connman, const std::atomic<bool>& interr
             // Determine transactions to relay
             if (fSendTrickle) {
                 // Produce a vector with all candidates for sending
-                std::vector<std::set<uint256>::iterator> vInvTx;
+                std::vector<uint256> vInvTx;
                 vInvTx.reserve(pto->setInventoryTxToSend.size());
-                for (std::set<uint256>::iterator it = pto->setInventoryTxToSend.begin(); it != pto->setInventoryTxToSend.end(); it++) {
-                    vInvTx.push_back(it);
+                for (std::set<uint256>::iterator it = pto->setInventoryTxToSend.begin(); it != pto->setInventoryTxToSend.end(); ) {
+                    vInvTx.push_back(*it);
+                    it = pto->setInventoryTxToSend.erase(it);
```

Comment:
Yes, I realized that I broke things temporarily. I will rearrange the commits so that is not the case anymore.